src/modules/85-cv/pages/health-source/common/MetricThresholds/Components/ThresholdCriteria.tsx

```diff
@@ -3,70 +3,111 @@ import { useFormikContext } from 'formik'
+  const { values: formValues, setFieldTouched, setFieldValue } = useFormikContext<AppDynamicsFomikFormInterface>()
 
   const { getString } = useStrings()
 
   const isAbsoluteSelected = criteriaType === MetricCriteriaValues.Absolute
 
-  const isGreaterThanSelected = criteriaPercentageType === PercentageType.GreaterThan
-  const isLessThanSelected = criteriaPercentageType === PercentageType.LessThan
+  const criteriaSpecValue = formValues[thresholdTypeName as ThresholdsPropertyNames][index].criteria.spec
 
-  const showGreaterThanInput = isAbsoluteSelected || isGreaterThanSelected
-  const showLessThanInput = isAbsoluteSelected || isLessThanSelected
+  const showGreaterThan = getIsShowGreaterThan(
+    criteriaType,
+    criteriaPercentageType as CriteriaThresholdValues,
+    criteriaSpecValue
+  )
+
+  const showLessThan = getIsShowLessThan(
+    criteriaType,
+    criteriaPercentageType as CriteriaThresholdValues,
+    criteriaSpecValue
+  )
 
   useEffect(() => {
     setFieldTouched(`${thresholdTypeName}.${index}.criteria.spec.${FieldName.METRIC_THRESHOLD_GREATER_THAN}`, true)
     setFieldTouched(`${thresholdTypeName}.${index}.criteria.spec.${FieldName.METRIC_THRESHOLD_LESS_THAN}`, true)
-  }, [index, setFieldTouched, thresholdTypeName, isAbsoluteSelected, criteriaPercentageType])
 
-  const handleCriteriaUpdate = (selectedValue: string): void => {
+    // This is to set criteriaPercentageType during edit
+    if (!isAbsoluteSelected && !criteriaPercentageType) {
+      setFieldValue(
+        `${thresholdTypeName}.${index}.criteria.${FieldName.METRIC_THRESHOLD_CRITERIA_PERCENTAGE_TYPE}`,
+        showGreaterThan ? PercentageType.GreaterThan : PercentageType.LessThan
+      )
+    }
+  }, [
+    index,
+    setFieldTouched,
+    thresholdTypeName,
+    isAbsoluteSelected,
+    criteriaPercentageType,
+    setFieldValue,
+    showGreaterThan
+  ])
+
+  const handleCriteriaUpdate = (selectedValue: MetricThresholdCriteria['type']): void => {
     if (selectedValue === MetricCriteriaValues.Absolute) {
       return void 0
     }
 
-    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-    // @ts-ignore
     const clonedThresholdValue = [...formValues[thresholdTypeName]]
 
     const updatedThresholds = { ...clonedThresholdValue[index] }
 
-    const criteriaDetails = updatedThresholds[FieldName.METRIC_THRESHOLD_CRITERIA]
+    const criteriaDetails = updatedThresholds.criteria
 
     // whenever the criteria value is changed to Percentage, Greater than value is set as default and Less than is set to undefined
     criteriaDetails.type = selectedValue
-    criteriaDetails[FieldName.METRIC_THRESHOLD_CRITERIA_PERCENTAGE_TYPE] = PercentageType.GreaterThan
-    criteriaDetails.spec[FieldName.METRIC_THRESHOLD_LESS_THAN] = undefined
+    criteriaDetails.criteriaPercentageType = PercentageType.GreaterThan
+
+    /* istanbul ignore next */
+    if (typeof criteriaDetails.spec === 'undefined') {
```

Comment:
Can we check for undefined values  this way ?

```suggestion
    if (criteriaDetails?.spec) {
```

Reply:
I want it more to be declarative and more explicit. So that if someone else works on the code later, they can get to know what is happening.

Reply:
And here I wanted to check whether `criteriaDetails.spec` (`spec`) is `undefined` or not. `criteriaDetails` value will be present always.

Reply:
ok , but this approach involves both a type lookup and a string comparison , which seems costly
Will keep this comment open feel free to pick up in next PR